aasemble/django/apps/buildsvc/pkgbuild/__init__.py

```diff
@@ -102,7 +102,11 @@ def checkout(self):
                  '--recursive',
                  '-b', self.build_record['source']['git_branch'],
                  self.builddir], logger=self.logger)
-        run_cmd(['git', 'reset', '--hard', self.build_record['sha']], cwd=self.builddir, logger=self.logger)
+        cmd = ['git', 'reset', '--hard']
+        build_id = self.build_record['sha']
+        if build_id is not None:
+            cmd.append(build_id)
+        run_cmd(cmd, cwd=self.builddir, logger=self.logger)
```

Comment:
If there's no sha, there's no point in doing the git reset.

Why is there no sha?